### PR TITLE
chore: add files field in package.json

### LINKED ISSUE #156: build: add src to .npmignore
### Describe the bug

I was releasing `6.0.0-alpha.3` and I noticed that the `src` folder is in the distributable package.

### Reproduction

Run `npm pack` and review the subsequent tarball.

### Additional Info

_No response_

### PR DESCRIPTION
**Pull request by MohdImran001**
Fixes #156 

@Shinigami92 please review

### CODE REVIEW

**Review by Shinigami92 — CHANGES_REQUESTED** [review]
Please delete `.npmignore`

File: package.json [hunk 1]
```diff
@@ -102,5 +102,17 @@
...
+  },
+  "files": [
+    "CHANGELOG.md",
+    "CHANGELOG_old.md",
+    "index.d.ts",
+    "index.js",
+    "index.ts",
+    "lib",
+    "locale",
+    "MIT-LICENSE.txt",
```
Comment (Shinigami92) [inline package.json:114]: LGTM, but we may want to wait on #157, so we don't miss that one

Comment (MohdImran001) [reply package.json:114]: Do I need to remove `MIT-LICENSE.txt` because `LICENSE` will be added regardless of being mentions in the `files` array

Comment (Shinigami92) [reply package.json:114]: You are right! Please just remove it :slightly_smiling_face:

**Comment by Shinigami92** [comment]
There is also a dist folder :thinking: 
@damienwebdev is that also needed? Or how does this folder work?

File: package.json [hunk 2]
```diff
@@ -102,5 +102,16 @@
...
     "vinyl-transform": "^1.0.0",
     "vitepress": "^0.21.4"
-  }
+  },
+  "files": [
+    "CHANGELOG.md",
+    "CHANGELOG_old.md",
+    "index.d.ts",
+    "index.js",
+    "index.ts",
```
Comment (Shinigami92) [inline package.json:111]: Please remove this file from package bundle
This will fix #176 / #177

**Review by Shinigami92 — CHANGES_REQUESTED** [review]

File: package.json [hunk 3]
```diff
@@ -102,5 +102,16 @@
     "vinyl-source-stream": "^2.0.0",
     "vinyl-transform": "^1.0.0",
     "vitepress": "^0.21.4"
-  }
+  },
+  "files": [
+    "CHANGELOG.md",
+    "CHANGELOG_old.md",
+    "index.d.ts",
```
Comment (Shinigami92) [inline package.json:109]:
```suggestion
    "dist",
    "index.d.ts",
```

**Review by Shinigami92 — APPROVED** [review]

**Review by prisis — APPROVED** [review]

**Review by import-brain — APPROVED** [review]

**Pull request merged**